### PR TITLE
Allow unlimited length for message conversion to string

### PR DESCRIPTION
**Pull request by boldorider4**

### CODE REVIEW

File: zmq.hpp [hunk 1]
```diff
@@ -538,8 +538,10 @@ class message_t
 
     /** Dump content to string. Ascii chars are readable, the rest is printed as hex.
          *  Probably ridiculously slow.
+         *  If zero is passed to limit_printable_length, the the string, the message
+         *  data will be printed in its entirety.
          */
-    std::string str() const
+    std::string str(const unsigned int limit_printable_length = 1000) const
```
Comment (sigiesec) [inline zmq.hpp]: size_t would be more consistent here

File: zmq.hpp [hunk 2]
```diff
@@ -538,8 +538,10 @@ class message_t
 
     /** Dump content to string. Ascii chars are readable, the rest is printed as hex.
          *  Probably ridiculously slow.
+         *  If zero is passed to limit_printable_length, the the string, the message
```
Comment (sigiesec) [inline zmq.hpp]: I am not sure what this should actually be. "the the string" can be left out entirely, but maybe there is something missing.

**Comment by gummif** [comment]
Can you resolve the conflict? @boldorider4

**Comment by gummif** [comment]
Closing this stale PR.

**Pull request closed without merging**